cloud/docker.go

```diff
@@ -83,17 +83,26 @@ func (m *dockerManager) SpawnHost(ctx context.Context, h *host.Host) (*host.Host
 	})
 
 	// Create container
-	if err := m.client.CreateContainer(ctx, parent, h.Id, h.Distro.User, settings); err != nil {
+	if err := m.client.CreateContainer(ctx, parentHost, h, settings); err != nil {
 		err = errors.Wrapf(err, "Failed to create container for host '%s'", hostIP)
 		grip.Error(err)
 		return nil, err
 	}
 
+	if err = h.SetAgentRevision(evergreen.BuildRevision); err != nil {
```

Comment:
This may or may not be feasible, but I feel like these 2 db interactions should be in the job that is calling the cloud manager, not the cloud manager itself (which should really only be to talk to docker). Is that doable without a bunch of refactoring?

Reply:
@john-m-liu This currently isn't feasible. The agent revision and host status need to be set after building the container image and before starting the container. Currently, all of this happens inside of SpawnHost in Docker.go, so we can't set these fields outside of the cloud provider in the provisioning-create-host job.

However, with the larger refactor we talked about earlier (downloading and provisioning the required images before calling SpawnHost, which should just create and start the container based on the image), this will definitely change.